cmd/snap-update-ns/main.go

```diff
@@ -162,6 +162,9 @@ func computeAndSaveChanges(snapName string) error {
 			}
 		}
 		if err != nil {
+			if change.Entry.XSnapdOrigin() == "layout" {
+				return err
+			}
```

Comment:
There is this comment up above this check for `err != nil`:
```
// NOTE: we may have done something even if Perform itself has failed.
// We need to collect synthesized changes and store them.
```

I am a fan of this commit in general and failing to run the snap if the layout fails. I wonder if the `err != nil` check should now be closer to the changePerform() and the comment adjusted? Perhaps not since, like you said in earlier commits, this is but one of 3 different types of mounts.... Since the ordering here is tricky and not immediately clear without the larger context, perhaps add a comment here?